### PR TITLE
README updates for introduction

### PR DESCRIPTION
**Pull request by pzharrington**
Added:

- README details for intro section, first single GPU section (reservation info, etc)
- Corrected some commands from single GPU performance section (no more `num_epochs`)
- Reverted submission script to 23.07 container version
- Disabled logging from ranks other than slurm rank 0

### CODE REVIEW

File: submit_pm.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #SBATCH --cpus-per-task 32
 #SBATCH --gpus-per-node 4
 #SBATCH --time=01:00:00
```
Comment (ShashankSubramanian) [inline submit_pm.sh:6]: should this be longer?

File: README.md [hunk 2]
```diff
@@ -1,12 +1,104 @@
...
+**Contents**
+* [Links](#links)
+* [Installation](#installation-and-setup)
+* [Model, data, and code overview](#model-data-and-training-code-overview)
+* [Single GPU training](#single-gpu-training)
+* [Single GPU performance](#single-gpu-performance-profiling-and-optimization)
+* [??](#distributed-gpu-training)
+* [??](#multi-gpu-performance-profiling-and-optimization)
+* [??](#putting-it-all-together)
+
```
Comment (ShashankSubramanian) [inline README.md:14]: + model parallel section before putting it all together

File: utils/data_loader.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def get_data_loader(params, files_pattern, distributed, train):
     else:
         return dataloader, dataset
 
-class GetDataset(Dataset):
+class ERA5Dataset(Dataset):
     def __init__(self, params, location, train):
```
Comment (ShashankSubramanian) [inline utils/data_loader.py:45]: Can you change this name convention in the dali es helper as well?

File: README.md (hunk 2 above)
Comment (pzharrington) [reply README.md:14]: I'm just leaving these links/etc blank until we merge in the final sections

Comment (pzharrington) [reply README.md:14]: But yeah I'll just rename the header tag now

File: utils/data_loader.py (hunk 3 above)
Comment (pzharrington) [reply utils/data_loader.py:45]: Did this in the latest commit -- can you double check it looks ok? (the DALI es helper is not used anywhere else right?)

File: utils/data_loader_dali.py
Comment (ShashankSubramanian) [inline utils/data_loader_dali.py:1]: Change in two places

**Comment by pzharrington** [comment]
Ok, this should be ready to go; it also incorporates model parallel readme updates from #5 (but added to main README rather than a separate file). So merging this can close out #5 as well. @ShashankSubramanian want to click the button? :)

**Comment by ShashankSubramanian** [comment]
Okay done

**Pull request merged**